### PR TITLE
Feature: Sharing links posted in channels to clickbait channel.

### PR DESCRIPTION
**Pull request by baali**
Feature: aggregate links to one channel.

### CODE REVIEW

File: parktain/main.py [hunk 1]
```diff
@@ -65,12 +65,23 @@ def checkins_reminder():
...
     message_log = Message(user_id=user, channel_id=channel, message=message, timestamp=datetime.now())
     session.add(message_log)
     session.commit()
 
+    # Check for presence of HyperLink in message
+    if user == bot.id:
+        return
+    for word in message.split():
+        try:
+            o = urlparse(word[1:-1])
```
Comment (punchagan) [inline parktain/main.py:79]: Why `word[1:-1]`? I don't understand it.

**Comment by punchagan** [comment]
We may want to keep more context about what the person says when they post the message? Like some kind of comment, etc?  Or may be link it to original message where it was posted? So, that people can see if there was any discussion and follow up?

File: parktain/main.py [hunk 2]
```diff
@@ -65,12 +65,23 @@ def checkins_reminder():
...
     bot.speak('Morning! What are you doing on {}!'.format(date), "#checkins")
 
-@bot.listen_for(lambda user, channel, message: True)
+@bot.listen_for(lambda user, channel, message: True, target_channel='clickbaits')
 def logger(user, channel, message):
     message_log = Message(user_id=user, channel_id=channel, message=message, timestamp=datetime.now())
     session.add(message_log)
     session.commit()
 
+    # Check for presence of HyperLink in message
```
Comment (punchagan) [inline parktain/main.py:74]: I'm leaning towards having a separate handler for this. Just keep it simple and let one function do one thing?

File: parktain/main.py [hunk 3]
```diff
@@ -65,12 +65,23 @@ def checkins_reminder():
...
+    if user == bot.id:
```
Comment (punchagan) [inline parktain/main.py:75]: I pushed a change to `gendo` to do this.  Or do you think some handlers should capture even bot messages? For instance the logger itself?

File: parktain/main.py [hunk 4]
```diff
@@ -65,12 +65,23 @@ def checkins_reminder():
     date = datetime.now().strftime('%d %B, %Y')
     bot.speak('Morning! What are you doing on {}!'.format(date), "#checkins")
 
-@bot.listen_for(lambda user, channel, message: True)
+@bot.listen_for(lambda user, channel, message: True, target_channel='clickbaits')
```
Comment (punchagan) [inline parktain/main.py:68]: I'm adding an `ignore_channel` flag.  What if someone posts a link directly to `clickbaits`? Do we want to ignore that or do we also notify that?

File: parktain/main.py (hunk 1 above)
Comment (baali) [inline parktain/main.py:79]: From slack, when there is a url in message I get url enclosed between <url>. So have to do this to skip them.
Example: 'http://qz.com/620076/beijing-is-banning-all-foreign-media-from-publishing-online-in-china/'

**Pull request closed without merging**